sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -244,17 +249,35 @@ public Aggregation toDruidAggregation(
     final AggregatorFactory theAggFactory;
     switch (args.size()) {
       case 1:
-        theAggFactory = aggregatorType.createAggregatorFactory(aggregatorName, fieldName, null, outputType, null);
+        theAggFactory = aggregatorType.createAggregatorFactory(aggregatorName, fieldName, null, outputType, null, true);
         break;
       case 2:
-        int maxStringBytes;
+        Integer maxStringBytes = getMaxStringBytes(rexNodes.get(1), aggregateCall, plannerContext);
+        if (maxStringBytes == null) {
+          return null;
+        }
+        theAggFactory = aggregatorType.createAggregatorFactory(
+            aggregatorName,
+            fieldName,
+            null,
+            outputType,
+            maxStringBytes.intValue(),
+            true
+        );
+        break;
+      case 3:
+        maxStringBytes = getMaxStringBytes(rexNodes.get(1), aggregateCall, plannerContext);
```

Comment:
nit: copy-paste - as the `createAggregatorFactory` will be called anyway with a `null` if its not specified; wouldn't it be better to just keep a variable and set its value when available?